doc/developer-guides/hld/ivshmem-hld.rst

```diff
@@ -181,3 +181,57 @@ post-launched VMs (VM1 and VM2).
+   - In UOS, the shared memory region can be accessed using ```mmap()``` of UIO device node. Please consider applying access control accordingly. Other complementary info can be found under:
+      * ```/sys/class/uio/uioX/device/resource2``` --> shared memory base address.
+      *  ```/sys/class/uio/uioX/device/config``` --> shared memory Size.
+   - For Linux based guest VMs, we recommend using the standard ```UIO``` and ```UIO_PCI_GENERIC``` drivers through the device node (Ex: ```/dev/uioX```)
+   - Reference: `AppArmour <https://wiki.ubuntuusers.de/AppArmor/>`_, `SELinux <https://selinuxproject.org/page/Main_Page>`_, `UIO driver-API <https://www.kernel.org/doc/html/v4.12/driver-api/uio-howto.html>`_
+
+
+3. **Crypto Support/Secure Applied Crypto**
+
+   - According to the application’s threat model and the defined assets that need to be shared securely, define the requirements for crypto algorithms. Those algorithms should enable operations like Authenticated Encryption/Decryption, Secure Key Exchange, True Random Number Generation, and Seed Extraction. In addition, consider the landscape of you attack surface and define the need for security engine (Ex: CSME) services.
+   - Don't implement your own crypto and use available Intel crypto libraries when applicable or other open source complaint libraries. (`Intel IPP <https://github.com/intel/ipp-crypto>`_, `TinyCrypt <https://01.org/tinycrypt>`_)
+   - Utilize the platform/kernel infrastructure and services (e.g. `vTPM <https://projectacrn.github.io/latest/developer-guides/hld/hld-security.html?highlight=vtpm>`_, `Kernel Crypto backend/APIs <https://www.kernel.org/doc/html/v5.4/crypto/index.html>`_ , `keyring subsystem <https://www.man7.org/linux/man-pages/man7/keyrings.7.html>`_, etc..).
+   - Depending on the used crypto key type and in case of requirements for key persistence across system/power management events, implement any necessary flows for key lifecycle management including wrapping, revocation and migration.
+   - Follow open source secure crypto coding guidelines for secure wrappers and marshalling data structures: `Secure Applied Crypto <https://github.com/veorq/cryptocoding>`_
+   - References: `NIST Crypto Standards and Guidelines <https://csrc.nist.gov/projects/cryptographic-standards-and-guidelines>`_, `OpenSSL <https://www.openssl.org/>`_
+
+
+4. **Applications Whitelisting**
+
+   - For use cases implemented in static environments for (Ex: Industrial/Automotive usages, make sure to follow application whitelist techniques and disable any 3rd party or native app stores.
+   - This mechanism can be chained with the access control policies to protect access to whitelisting rules/configuration files. Please refer to opensource or implement your custom solution.
```

Comment:
While polite, we don't use "please" in the documentation.

Reply:
done